### PR TITLE
minor refactoring

### PR DESCRIPTION
**Pull request by AchintyaX**

### CODE REVIEW

File: dialogy/plugins/text/list_search_plugin/__init__.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def search_regex(
                     final_match = match_value
                     match_span = result.span()
         if final_match:
-            return (match_text, entity_type, final_match, match_span, float(0))
+            return (match_text, entity_type, final_match, match_span, float(1))
```
Comment (ltbringer) [inline dialogy/plugins/text/list_search_plugin/__init__.py:207]: Use 0.0, 1.0 instead of `float(0)`, `float(1)`.

File: dialogy/plugins/text/list_search_plugin/__init__.py [hunk 2]
```diff
@@ -307,14 +307,16 @@ def get_entities(self, transcripts: List[str], lang: str) -> List[BaseEntity]:
         entities: List[BaseEntity] = []
 
         for i, matches_on_transcript in enumerate(matches_on_transcripts):
-            for text, label, value, span, Score in matches_on_transcript:
+            for text, label, value, span, score in matches_on_transcript:
+                if score == float(0):
```
Comment (ltbringer) [inline dialogy/plugins/text/list_search_plugin/__init__.py:311]: `int(score)` is better than `float(0)`

**Review by ltbringer — CHANGES_REQUESTED** [review]

File: dialogy/plugins/text/list_search_plugin/__init__.py (hunk 2 above)
Comment (AchintyaX) [reply dialogy/plugins/text/list_search_plugin/__init__.py:311]: if we do `int(score)` then it would always be 0, since `0<=score<=1`

File: dialogy/plugins/text/list_search_plugin/__init__.py (hunk 1 above)
Comment (AchintyaX) [reply dialogy/plugins/text/list_search_plugin/__init__.py:207]: implementing this change

File: dialogy/plugins/text/list_search_plugin/__init__.py (hunk 2 above)
Comment (ltbringer) [reply dialogy/plugins/text/list_search_plugin/__init__.py:311]: then let's use 0.0 for score? also that should still equate if we were using a 0 too.

File: dialogy/plugins/text/list_search_plugin/__init__.py [hunk 3]
```diff
@@ -204,8 +204,8 @@ def search_regex(
                     final_match = match_value
                     match_span = result.span()
         if final_match:
-            return (match_text, entity_type, final_match, match_span, float(0))
-        return ("", entity_type, "", (0, 0), float(0))
+            return (match_text, entity_type, final_match, match_span, 1.0)
```
Comment (ltbringer) [inline dialogy/plugins/text/list_search_plugin/__init__.py:207]: is this change intentional? float(0) -> 1.0?

Comment (AchintyaX) [reply dialogy/plugins/text/list_search_plugin/__init__.py:207]: yes, idea is that since the regex match is 100% match we should keep the score as 1.0, earlier it was 0. but we were not removing the matches with score 0 earlier now we are hence this is something I did intentionally
